### PR TITLE
Documentation for Telemetry

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: #384 

Add documentation for the telemetry implementation.

### CODE REVIEW

File: Documentation/CorePackages/Telemetry.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Telemetry
+
+### What is it?
+
+Component responsible for notifying [Helix](https://helix.dot.net/) of build start and finish events. Essentially, it interacts with the Helix API to notify that a new build is starting or have just finished.
```
Comment (alexperovich) [inline Documentation/CorePackages/Telemetry.md]: type: have -> has

File: Documentation/CorePackages/Telemetry.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+The current implementation for sending start events can be found [here](../../eng/common/templates/steps/telemetry-start.yml) and, for build finish events, it's available [here](../../eng/common/templates/steps/telemetry-end.yml).
+
+### When is it run?
+
+If telemetry is enabled on the build definition it will be executed for every build that execute.
+
+### What is the result?
+
+As a result of this interaction the status of the build will be displayed on the [Mission Control](https://mc.dot.net/) web site. Also, you'll be able to see the build status history for the project.
```
[the remark this answers is not in the document]
Comment (alexperovich) [reply Documentation/CorePackages/Telemetry.md]: I don't think you need anything actually. Just start with "The status of the build will be dis..."

File: Documentation/CorePackages/Telemetry.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+
+### Will I use it automatically when onboarding?
+
+Yes. As part of the onboarding process in Arcade you will copy the `eng` folder from the root of the repository. The YAML build definition scripts on that folder take care of setting up the telemetry infrastructure.
```
Comment (alexperovich) [inline Documentation/CorePackages/Telemetry.md]: The entire `eng` folder should not ever be copied. Just the `eng\common` folder. The contents of the `eng` folder are repository specific.

Comment (alexperovich) [inline Documentation/CorePackages/Telemetry.md]: typo: "build definition scripts on that folder" -> "build definition scripts in that folder"

File: Documentation/CorePackages/Telemetry.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (JohnTortugo) [reply Documentation/CorePackages/Telemetry.md]: I did some rewording on this section.

> Actually, there's some additional things ...

I think creating another document with guidelines on how to get the project tracked on MC would be better.

File: Documentation/CorePackages/Telemetry.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+### When does it run?
+
+If telemetry is enabled on the build definition, telemetry events will be sent for every queued build.
+
+### What is the result?
+
+Executing these telemetry scripts during the build is part of the requisites to see your build results on the [Mission Control](https://mc.dot.net/) web site. Also, you'll be able to see the build status history for the project.
+
+
```
Comment (chcosta) [inline Documentation/CorePackages/Telemetry.md]: nit: extra space

File: Documentation/CorePackages/Telemetry.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+### Will the repo automatically the telemetry when onboarding?
```
Comment (chcosta) [inline Documentation/CorePackages/Telemetry.md]: Will the repo automatically **report** the telemetry when onboarding?

File: Documentation/CorePackages/Telemetry.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+Yes. As part of the onboarding process in Arcade you're supposed to copy the `eng\common` folder from the root of the repository and reference the `eng\common\templates\phases\base.yml` file on your build definitions. This will plug in the use of the telemetry scripts.
```
Comment (chcosta) [inline Documentation/CorePackages/Telemetry.md]: It's not automatic, you also have to set `enableTelemetry` to `true` and specify a couple of variables - https://github.com/dotnet/arcade/blob/master/eng/common/templates/phases/base.yml#L22-L27

File: Documentation/CorePackages/Telemetry.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+The current implementation for sending start events can be found [here](../../eng/common/templates/steps/telemetry-start.yml) and, for build finish events, it's available [here](../../eng/common/templates/steps/telemetry-end.yml).
+
+### When does it run?
+
+If telemetry is enabled on the build definition, telemetry events will be sent for every queued build.
+
+### What is the result?
+
+Executing these telemetry scripts during the build is part of the requisites to see your build results on the [Mission Control](https://mc.dot.net/) web site. Also, you'll be able to see the build status history for the project.
```
Comment (chcosta) [inline Documentation/CorePackages/Telemetry.md]: requisites -> prerequisites

File: Documentation/CorePackages/Telemetry.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+### What is it?
+
+Component responsible for notifying [Helix](https://helix.dot.net/) of build start and finish events. Essentially, it interacts with the Helix API to notify that a new build is starting or has just finished.
+
+The current implementation for sending start events can be found [here](../../eng/common/templates/steps/telemetry-start.yml) and, for build finish events, it's available [here](../../eng/common/templates/steps/telemetry-end.yml).
+
+### When does it run?
+
+If telemetry is enabled on the build definition, telemetry events will be sent for every queued build.
```
Comment (chcosta) [inline Documentation/CorePackages/Telemetry.md]: This is a bit misleading as it implies that there is a setting in the VSTS build definition, but below you say that telemetry is enabled via the YAMl which is loaded by the build definition.

File: Documentation/CorePackages/Telemetry.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chcosta) [reply Documentation/CorePackages/Telemetry.md]: Who's creating that document?

[the remark this answers is not in the document]
Comment (JohnTortugo) [reply Documentation/CorePackages/Telemetry.md]: Created this issue #609

**Comment by chcosta** [comment]
I don't really have any major feedback apart from the sentence fragment "...you're supposed to copy the..." being a bit too polite as it's a requirement, not an optional step.

Overall, this doc feels very light on details, and if that's the intent then great.  I could, however, imagine this providing a bit more value if it highlighted the value of telemetry and some examples of how we use it, what's collected, benefits to customers, etc...   Also, I still think it would be nice if this doc could link to something related to enabling views on mission control because if I were to find this document I'd be a little frustrated that it dead ends that scenario which I think is important.  All of that feedback is dependent on what the goal of this document is though.

**Comment by JohnTortugo** [comment]
As per the issue description I had understood that the goal of this doc was to provide "documentation on how to consume the telemetry scripts". But, IMO I think that what you're suggesting is great. I'll work to collect more information and add a new section on the document, something like "What does it provide?"

**Comment by JohnTortugo** [comment]
I've pushed a new version of the document. I'm waiting to talk with Chris Bohm so I can add some more info about the benefits that Mission Control brings in to our customers.

**Comment by JohnTortugo** [comment]
Hey, I just pushed a new version where I add a bit of info about the benefits of using these scripts to collect telemetry. Can you please take a look? I'd like to merge this by the end of the day today. Thanks.

**Pull request merged**